docs/docsite/rst/community/development_process.rst

```diff
@@ -33,29 +34,26 @@ After the pull request submitted to Ansible for the ``devel`` branch is
 accepted and merged, the following instructions will help you create a
 pull request to backport the change to a previous stable branch.
 
-.. note::
-
-    These instructions assume that ``stable-2.5`` is the targeted release
-    branch for the backport.
+We do **not** backport features.
```

Comment:
@acozine Feel free to reword this and merge once you are happy

Reply:
Looks good, thanks @gundalow for the changes and @vladimir-mencl-eresearch for pointing out an area where we could improve!